pkg/threescale/connectors/backend/cached_backend.go

```diff
@@ -123,13 +131,13 @@ func (cb CachedBackend) handleCacheMiss(req AuthRepRequest, cacheKey string, c *
 
 // handleCachedRead works on a copy of the passed CacheValue and will inform the caller if limits have
 // been breached in the provided affected metrics
-func (cb CachedBackend) handleCachedRead(value CacheValue, affectedMetrics backend.Metrics) (CacheValue, bool) {
+func (cb CachedBackend) handleCachedRead(value CacheValue, appID string, affectedMetrics backend.Metrics) (CacheValue, bool) {
 	copyCache := cloneCacheValue(value)
 	commitChanges := true
 
 out:
 	for metric, incrementBy := range affectedMetrics {
-		cachedValue, ok := copyCache.LimitCounter[metric]
+		cachedValue, ok := copyCache[appID].LimitCounter[metric]
```

Comment:
perhaps we can pass in `appID` to `cloneCacheValue` and have `copyCache` already be the `copyCache[appID]` used here?